Library/Homebrew/cmd/--version.rb

```diff
@@ -5,10 +5,9 @@ module Homebrew
   module_function
 
   def __version
-    # As a special case, `--version` is implemented directly in `brew.rb`. This
-    # file merely serves as a container for the documentation. It also catches
-    # the case where running `brew --version` with additional arguments would
-    # produce a rather cryptic message about a non-existent `--version` command.
-    raise UsageError
+    raise UsageError, "This command does not take arguments." unless ARGV.empty?
```

Comment:
Could maybe use the same form here as e.g. `brew pull` i.e. the new option parser stuff?

Reply:
I think this is better, since a `UsageError` shows the command's help along with the error message.

Reply:
@reitermarkus I disagree; I'd rather see `Homebrew::CLI::Parser.parse` extended to output the command's help and have more consistent behaviour for now.